pkg/uid/from_etc.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// For how the uidshift and uidcount are generated please check:
+// http://cgit.freedesktop.org/systemd/systemd/commit/?id=03cfe0d51499e86b1573d1
+
+package uid
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/coreos/rkt/pkg/group"
+	"github.com/coreos/rkt/pkg/passwd"
+	"github.com/hashicorp/errwrap"
+)
+
+// FromEtc is the struct that generates a uid/gid by parsing etc/passwd,
+// and etc/group relative from the given Root path looking for the given Name.
+type FromEtc struct {
+	Root string
+	Name string
+}
+
+func (e FromEtc) UID() (int, error) {
+	uid_, err := passwd.LookupUidFromFile(
+		e.Name,
+		filepath.Join(e.Root, "etc/passwd"),
+	)
+
+	if err != nil {
+		return -1, errwrap.Wrap(fmt.Errorf("cannot lookup user %q", e.Name), err)
+	}
+
+	return uid_, nil
+}
+
+func (e FromEtc) GID() (int, error) {
+	gid_, err := group.LookupGidFromFile(
```

Comment:
@tmrts @jellonek re package name: if in doubt, look upstream; https://godoc.org/os/user solves similar issues hence my suggestion is to call this package `user`.

@tmrts re `passwd` as the src for GIDs: I refactored the implementation from https://github.com/coreos/rkt/blob/master/stage1/init/common/pod.go#L607-L667 to make it reusable in the fly favor. If we want to change the functionality, I suggest to address this in a separate issue and separate PR. For this I'd leave the actual functionality untouched.

@tmrts: The `uid_` is to prevent a collision with the package name itself, if we call it "user", those become obsolete.